lxterminal: init at 0.3.1

velovix opened the pull request:
###### Motivation for this change

LXTerminal is the standard terminal emulator for LXDE, so LXDE users would expect it to exist. Subjectively, it's also a good desktop-agnostic terminal that I use even though I don't use LXDE.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

velovix committed:
lxterminal: init at 0.3.1

Ericson2314 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -359,6 +359,10 @@ with pkgs;
 
   findXMLCatalogs = makeSetupHook { } ../build-support/setup-hooks/find-xml-catalogs.sh;
 
+  buildSingleXMLCatalog = makeSetupHook {
```
This now has a `name` parameter we should be using.

Ericson2314 commented on pkgs/build-support/setup-hooks/build-single-xml-catalog.sh:
```diff
@@ -0,0 +1,21 @@
+    echo '</catalog>' >> catalog.xml
+}
+
+if [ -z "$buildSingleXMLCatalogHookDone" ]; then
```
You don't want this. Environment Hooks are functions applied to every dependency, but `buildSingleXMLCatalog` takes no argument. Just remove the whole `if` block and call the function manually (e.g. in `preConfigure` of the package you are packaging).

Ericson2314 commented on pkgs/build-support/setup-hooks/build-single-xml-catalog.sh:
```diff
@@ -0,0 +1,21 @@
+# Creates a single XML catalog that references the catalogs found by
+# findXMLCatalogs.
+# Useful for situations where only one catalog is expected.
+buildSingleXMLCatalog() {
```
Probably better if this takes a single argument which is the output file produced.

Ericson2314 reviewed (changes requested):
Oops sorry that was supposed to be a "request changes" review.

jtojnar commented on pkgs/applications/misc/lxterminal/default.nix:
```diff
@@ -0,0 +1,44 @@
+
+  configureFlags = [
+    "--enable-man"
+    "--with-xml-catalog=../catalog.xml" # Generated by buildSingleXMLCatalog
```
Alternately, it is possible to patch the m4 file to autodetect the catalog from the variable, see for example [`gnome3.cheese`](https://github.com/NixOS/nixpkgs/blob/0e7358002679ad7f35d2bd99977b9f0e403d4de2/pkgs/desktops/gnome-3/apps/cheese/default.nix#L23-L25).

velovix replied:
Good to know, that makes things a lot easier.

jtojnar replied:
The only issue is, the macro is in `acinclude.m4` instead of the usual `m4/gtkdoc_jh_check_xml_catalog.m4`.

velovix replied:
Right, and the m4 file has different contents as well. I ended up generating [my own patch](https://github.com/velovix/nixpkgs/blob/a5880ace7fbc4359d3f0d8621fff16a0977283a1/pkgs/applications/misc/lxterminal/respect-xml-catalog-files-var.patch).

velovix committed:
lxterminal: patch m4 to respect XML_CATALOG_FILES

The patch removes the need for a separate script to combine existing
catalog files, making the package simpler.

velovix commented:
Following the advice of @jtojnar I've removed the XML catalog combining setup hook in favor of a solution that patches the m4 file to respect the XML_CATALOG_FILES environment variable.

@Ericson2314: This sidesteps the problems you had found with the setup hook solution. Please take a look.

Ericson2314 reviewed (approved):
Yeah that was a great idea. Looks good!
